src/Deserializers/AliasGroupDeserializer.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Wikibase\DataModel\Deserializers;
+
+use Deserializers\Deserializer;
+use Deserializers\Exceptions\DeserializationException;
+use Deserializers\Exceptions\InvalidAttributeException;
+use Deserializers\Exceptions\MissingAttributeException;
+use Wikibase\DataModel\Term\AliasGroup;
+
+/**
+ * @since 1.5
+ *
+ * @author Adam Shorland
+ */
+class AliasGroupDeserializer implements Deserializer {
+
+	public function __construct() {
+	}
+
+	/**
+	 * @param mixed $serialization
+	 *
+	 * @return AliasGroup
+	 * @throws DeserializationException
+	 */
+	public function deserialize( $serialization ) {
+		$this->assertCanDeserialize( $serialization );
+
+		return $this->getDeserialized( $serialization );
+	}
+
+	/**
+	 * @param array $serialization
+	 *
+	 * @return AliasGroup
+	 */
+	private function getDeserialized( $serialization ) {
+		foreach ( $serialization as $requestedLanguage => $aliasesPerLanguageSerialization ) {
+			if ( !is_array( $aliasesPerLanguageSerialization ) ) {
+				throw new DeserializationException( "Aliases serialization should be an array of array" );
+			}
+
+			$aliases = array();
+
+			foreach ( $aliasesPerLanguageSerialization as $aliasSerialization ) {
+				$this->assertIsValidValueSerialization( $aliasSerialization, $requestedLanguage );
+				$aliases[] = $aliasSerialization['value'];
+			}
+
+			return new AliasGroup(
+				$requestedLanguage,
+				$aliases
+			);
+		}
+	}
+
+	/**
+	 * @param array $serialization
+	 */
+	private function assertCanDeserialize( $serialization ) {
+		if ( !is_array( $serialization ) ) {
+			throw new DeserializationException( 'The alias group serialization should be an array' );
+		}
+		if( count( $serialization ) > 1 ) {
```

Comment:
new line before `if` and space after `if`. I'd even suggest to only use one `if` clause covering both and please use `empty` instead of `count`.